src/Dialogs/Imap/ImapDialog.vala

```diff
@@ -558,7 +561,11 @@ public class OnlineAccounts.ImapDialog : Hdy.Window {
         unowned var transport_auth_extension = (E.SourceAuthentication) transport_source.get_extension (E.SOURCE_EXTENSION_AUTHENTICATION);
         transport_auth_extension.host = smtp_server_entry.text;
         transport_auth_extension.port = (uint) smtp_port_spin.value;
-        transport_auth_extension.user = smtp_username_entry.text;
+        if (use_imap_credentials.active) {
```

Comment:
Is this check necessary? If we don't want to use imap creds, we use smtp user which makes sense. If we *do* use imap creds the smtp user gets updated already so we could still use it, no?

Reply:
There is a subtle situation where it does not get updated properly:

- User has 'Use IMAP Credentials' turned OFF
- User changes the input of the SMTP username field
- User changes their mind and turns the 'Use IMAP Credentials' back ON
- The SMTP field now does not get updated with whatever is in the IMAP username field, until the user edits the IMAP username field again.

This check makes sure that we actually, definitively, use the IMAP username field's data for SMTP if this option is checked.

Reply:
I noted in the description that this line change solves the actual problem; the other change (syncing the fields) is just an affordance for the user.

Reply:
I see, yeah, the only other way would be to use a new `use_imap_credentials.bind_property` and have it update as it needs to when the checkbox changes. But unless you like that more I say this is fine and it can be merged.

Reply:
Thanks! Let’s keep it like this.